### DISCUSSION TITLE
Assistance Needed for Running ERA5 Data Download in UMEP Plugin | QGIS 3.34.11

### DISCUSSION THREAD

**Discussion by buritilg** [UMEP-dev/UMEP#658] [General]
I am encountering an issue while trying to run the ERA5 data download in the UMEP plugin for QGIS 3.34.11. Specifically, I receive the following error message:
`
AttributeError: 'NoneType' object has no attribute 'write'
`
The error seems to occur within the `tqdm` package while it attempts to handle a progress bar during the file download process.

Below is a snippet of the traceback for reference:
`
Traceback (most recent call last):
  File "C:\Users\laraa\AppData\Roaming\Python\Python312\site-packages\tqdm\std.py", line 452, in fp_write
    fp.write(str(s))
AttributeError: 'NoneType' object has no attribute 'write'
`
I succeeded in connecting my CDS API. I can see the UMEP plugin requests on my profile, but I am not able to generate any meteorological output with the UMEP plugin yet. 
![image](https://github.com/user-attachments/assets/c71dac3e-ae53-46d7-848c-be16cd4f03fd)

What are all the Python packages needed to run UMEP pre-processor?
I would appreciate any guidance or steps to resolve this issue. Thank you in advance for your help!

**Comment by biglimp** [comment]
The CDS API sometimes suffer from high traffic which makes download extremely slow. You can either try again or try to download from other sources e.g. www.shinyweatherdata.com or Google Colab/Google Cloud.

**Reply by I-T-H** [reply]
I also encountered this problem. I have hundreds of ERA5 files to download, but each one takes several hours or even more to download. Can you provide tutorials on the above two methods?
